File: Bucles/Ejercicio16-Bucles.cpp
/*16. Realice un programa que calcule la descomposición en factores primos de un número entero.

Por ejemplo: 20 = 2 * 2 * 5 */

#include<iostream>
#include<stdlib.h>

using namespace std;

// versión del maestro
int main (){
    int numero;
     cout<<"Digite un numero: "; 
     cin>>numero;

    for(int i=2;numero>1;i++){
        while(numero%i==0){
            cout<<i<<" ";
            numero = numero /i;
        }
    }
    cout<<"\n";
    system("pause");
    return 0;
}

File: Comparativa Python/Ejercicio16-Bucles.py
"""
16. Realice un programa que calcule la descomposición en factores primos de
un número entero.

Por ejemplo: 20 = 2 * 2 * 5
"""
n = int(input('Digite un número: '))

for i in range(2, n, 1):
    while n % i == 0:
        print(i, ' ', end="")
        n = n / i
